clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -109,6 +113,7 @@
     private AtomicBoolean asyncCommitFenced;
     private ConsumerGroupMetadata groupMetadata;
     private final boolean throwOnFetchStableOffsetsUnsupported;
+    private volatile boolean batchFetchOffsets = true;
```

Comment:
I can see this is based on the pattern used in the recent changes to FindCoordinator. But I don't see the point of it. If one broker that is the coordinator at any point in time wasn't upgraded, we switch to non-batched mode and stay there forever until the consumer is restarted. Given that consumers don't need batching, it seems to me that we could avoid any changes in consumers and just update `OffsetFetchRequest` to do the right thing based on versions. @dajac Since you reviewed the PR for FindCoordinators, what do you think?